src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
             {
                 physicalLocation = new PhysicalLocationVersionOne
                 {
-                    Region = CreateRegion(v2Region)
+                    Region = CreateRegion(v2Region, null)
                 };
             }
 
             return physicalLocation;
         }
 
-        internal RegionVersionOne CreateRegion(Region v2Region)
+        internal RegionVersionOne CreateRegion(Region v2Region, Uri uri)
         {
             RegionVersionOne region = null;
 
-            if (v2Region != null && (v2Region.StartColumn > 0 ||
-                                     v2Region.StartLine > 0 ||
-                                     v2Region.EndColumn > 0 ||
-                                     v2Region.EndLine > 0 ||
-                                     v2Region.Length > 0 ||
-                                     v2Region.Offset > 0))
-            {
-                region = new RegionVersionOne
-                {
-                    EndColumn = v2Region.EndColumn,
-                    EndLine = v2Region.EndLine,
-                    Length = v2Region.Length,
-                    Offset = v2Region.Offset,
-                    StartColumn = v2Region.StartColumn,
-                    StartLine = v2Region.StartLine
-                };
+            if (v2Region != null)
+            {
+                region = new RegionVersionOne();
+
+                if (v2Region.StartLine > 0 ||
+                    v2Region.EndLine > 0 ||
+                    v2Region.StartColumn > 0 ||
+                    v2Region.EndColumn > 0 ||
+                    v2Region.CharOffset > 0 ||
+                    v2Region.CharLength > 0)
+                {
+                    if (v2Region.StartLine > 0)
+                    {
+                        // The start of the region is described by line/column
+                        region.StartLine = v2Region.StartLine;
+                        region.StartColumn = v2Region.StartColumn > 0
+                            ? v2Region.StartColumn
+                            : 1;
+                    }
+                    else
+                    {
+                        // The start of the region is described by character offset
+
+                        if (v2Region.CharOffset == 0)
+                        {
+                            region.Offset = 0;
+                        }
+                        else
+                        {
+                            // Try to get the byte offset using the file encoding and contents
+                            region.Offset = GetRegionByteOffset(v2Region, uri);
```

Comment:
You don't need to pass the whole region; the only thing it uses is charOffset. So I suggest you rename the method: `ConvertCharOffsetToByteOffset(v2Region.CharOffset, uri)`

Reply:
... but see my next comment...

Reply:
Yup, changed